lib/AST/ASTPrinter.cpp

```diff
@@ -673,8 +674,18 @@ class PrintAST : public ASTVisitor<PrintAST> {
           M, cast<ValueDecl>(Current));
       }
 
-      T = T.subst(subMap,
-                  SubstFlags::DesugarMemberTypes | SubstFlags::UseErrorType);
+      SubstOptions substOptions = SubstFlags::UseErrorType;
+      if (Options.ForceKeepTypealiasTypeInTransformation)
+        substOptions |= SubstFlags::ForceKeepTypealias;
+      else
+        substOptions |= SubstFlags::DesugarMemberTypes;
+
+      T = T.subst(subMap, substOptions);
+
+      if (auto nominal = CurrentType->getNominalOrBoundGenericNominal())
```

Comment:
Two questions:

- When does CurrentType not have a nominal type?
- If it is a nominal type, doesn't this lose its generic arguments?